File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/token/SigningKeyRepository.java
package br.com.deltaglobalbank.identity.domain.token;

import java.util.List;
import java.util.UUID;

public interface SigningKeyRepository {
    SigningKey findById(UUID id);

    SigningKey findByKid(String kid);

    List<SigningKey> findAllByStatus(SigningKeyStatus status);

    SigningKey findFirstActive();

    SigningKey save(SigningKey signingKey);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/token/SigningKeyRepository.kt
package br.com.deltaglobalbank.identity.domain.token

import java.util.UUID

interface SigningKeyRepository {
    fun findById(id: UUID): SigningKey?
    fun findByKid(kid: String): SigningKey?
    fun findAllByStatus(status: SigningKeyStatus): List<SigningKey>
    fun findFirstActive(): SigningKey?
    fun save(signingKey: SigningKey): SigningKey
}
